src/promptflow/promptflow/executor/_service/apis/tool.py

```diff
@@ -0,0 +1,52 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+from fastapi import APIRouter
+
+from promptflow._core._errors import NoToolTypeDefined
+from promptflow._core.tool_meta_generator import generate_tool_meta_dict_by_file
+from promptflow._core.tools_manager import collect_package_tools
+from promptflow._utils.context_utils import _change_working_dir, inject_sys_path
+from promptflow.contracts.tool import ToolType
+from promptflow.executor._service._errors import GenerateMetaTimeout
+from promptflow.executor._service.contracts.tool_request import ToolMetaRequest
+from promptflow.executor._service.utils.process_utils import invoke_function_in_process
+from promptflow.executor._service.utils.service_utils import generate_error_response
+
+router = APIRouter(prefix="/tool")
+
+
+@router.get("/package_tools")
+async def list_package_tools():
+    return collect_package_tools()
+
+
+@router.post("/meta")
+async def gen_tool_meta(request: ToolMetaRequest):
+    result = await invoke_function_in_process(request, {}, gen_meta, is_async_call=False)
```

Comment:
Add try catch for timeout exception(ExecutionTimeoutError) here.

Reply:
Added.